core/src/ast/docs.rs

```diff
@@ -34,74 +34,158 @@ impl Docs {
+                            if let Some(ident) = display.get_ident() {
+                                let s = ident.to_string();
+                                match &*s {
+                                    "normal" => RustLinkDisplay::Normal,
+                                    "compact" => RustLinkDisplay::Compact,
+                                    "hidden" => RustLinkDisplay::Hidden,
+                                    _ => panic!("#[diplomat::rust_link()]'s third argument must be `normal`, `compact`, \
+                                                 or `hidden`, found: {}", display.to_token_stream()),
+                                }
+                            } else {
+                                panic!("#[diplomat::rust_link()]'s third argument must be a path, found: {}", display.to_token_stream());
+                            }
+                        } else {
+                            RustLinkDisplay::Normal
+                        };
+                        vec.push(RustLink {
+                            path: Path::from_syn(path),
+                            typ: match typ.to_token_stream().to_string().as_str() {
+                                "Struct" => DocType::Struct,
+                                "StructField" => DocType::StructField,
+                                "Enum" => DocType::Enum,
+                                "EnumVariant" => DocType::EnumVariant,
+                                "EnumVariantField" => DocType::EnumVariantField,
+                                "Trait" => DocType::Trait,
+                                "FnInStruct" => DocType::FnInStruct,
+                                "FnInEnum" => DocType::FnInEnum,
+                                "FnInTrait" => DocType::FnInTrait,
+                                "DefaultFnInTrait" => DocType::DefaultFnInTrait,
+                                "Fn" => DocType::Fn,
+                                "Mod" => DocType::Mod,
+                                "Constant" => DocType::Constant,
+                                "Macro" => DocType::Macro,
+                                x => panic!("Invalid doc type {:?}", x),
+                            },
+                            display,
+                        });
+                        continue;
+                    };
                 }
                 panic!("Malformed attribute: {}", attr.to_token_stream());
             }
         }
-        None
+        vec
     }
 
     pub fn is_empty(&self) -> bool {
-        self.0.is_empty() && self.1.is_none()
+        self.0.is_empty() && self.1.is_empty()
     }
 
-    pub fn to_markdown(&self, docs_url_gen: &DocsUrlGenerator) -> String {
+    /// Convert to markdown
+    ///
+    /// `in_rst` will avoid using complicated formatting combinations disallowed in ReStructuredText
+    pub fn to_markdown(&self, docs_url_gen: &DocsUrlGenerator, in_rst: bool) -> String {
```

Comment:
Nit: booleans in function parameters are not readable and not extensible; use an enum or options bag instead